app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue

```diff
@@ -53,19 +68,26 @@ const escapeString = (str: string): string => {
 }
 const makeArgsList = (args: string[]) => '[' + args.map(escapeString).join(', ') + ']'
 
-const selfArgumentExprId = computed<Opt<ExprId>>(() => {
-  const analyzed = ArgumentApplication.Interpret(props.input, true)
+const selfArgumentAstId = computed<Opt<ExprId>>(() => {
+  const analyzed = interpretCall(props.input.ast, true)
   if (analyzed.kind === 'infix') {
     return analyzed.lhs?.exprId
   } else {
-    return analyzed.args[0]?.argument.exprId
+    const knownArguments = methodCallInfo.value?.suggestion?.arguments
+    const selfArgument =
+      knownArguments?.[0]?.name === 'self'
+        ? getAccessOprSubject(analyzed.func)
+        : analyzed.args[0]?.argument
+    return selfArgument?.exprId
   }
 })
 
 const visualizationConfig = computed<Opt<NodeVisualizationConfiguration>>(() => {
-  const tree = props.input
-  const expressionId = selfArgumentExprId.value
-  const astId = tree.exprId
+  // If we inherit dynamic config, there is no point in attaching visualization.
+  if (props.input.dynamicConfig) return null
```

Comment:
Is it correct? If we have a function call in one of the arguments of another function call, we will receive a dynamic config for the outer call only.

Reply:
At this point, the `dynamicConfig` property is scoped to this particular argument value.

Reply:
The dynamic config for outer call could have quite a detailed config for its argument - for example, the Single_Choice specifies config for parameters of chosen call. I didn't want to make some complex merging system yet. Let's see how often we will be affected.